src/http.rs

```diff
@@ -205,7 +205,16 @@ pub fn build_tls_connector(
     let settings = tls_settings.tls().cloned();
     https.set_callback(move |c, _uri| {
         if let Some(settings) = &settings {
-            settings.apply_connect_configuration(c);
+            match settings.apply_connect_configuration(c) {
+                Ok(()) => (),
+                Err(error) => {
+                    error
+                        .source()
+                        .expect("was expecting to have a source in tlsError variant returned from apply_connect_configuration(). See SetSniSnafu.")
+                        .downcast_ref::<openssl::error::ErrorStack>()
+                        .expect("was expecting the source error to be of type openssl::error::ErrorStack from apply_connect_configuration(). See SetSniSnafu.");
+                }
+            }
```

Comment:
Thanks for adding the `expect`s. I'm not a huge fan of asserting that a specific enum variant from a Snafu error was returned. I tried to refactor this but wasn't able to push to this branch. Could you apply this patch?

```diff
9a5fabeb0e - (HEAD -> anil-db/support-sni) Refactor to get rid of `expect`s (23 seconds ago) <Jesse Szwedko>
diff --git a/lib/vector-core/src/tls/mod.rs b/lib/vector-core/src/tls/mod.rs
index 96b9253f4f..9e065cb212 100644
--- a/lib/vector-core/src/tls/mod.rs
+++ b/lib/vector-core/src/tls/mod.rs
@@ -191,7 +191,9 @@ fn tls_connector(settings: &MaybeTlsSettings) -> Result<ConnectConfiguration> {
         .context(TlsBuildConnectorSnafu)?;
     let tls_setting = settings.tls().cloned();
     if let Some(tls_setting) = &tls_setting {
-        tls_setting.apply_connect_configuration(&mut configure)?;
+        tls_setting
+            .apply_connect_configuration(&mut configure)
+            .context(SetSniSnafu)?;
     }
     Ok(configure)
 }
diff --git a/lib/vector-core/src/tls/settings.rs b/lib/vector-core/src/tls/settings.rs
index 1e68563011..1dcdc932b8 100644
--- a/lib/vector-core/src/tls/settings.rs
+++ b/lib/vector-core/src/tls/settings.rs
@@ -20,8 +20,8 @@ use super::{
     AddCertToStoreSnafu, AddExtraChainCertSnafu, CaStackPushSnafu, DerExportSnafu,
     EncodeAlpnProtocolsSnafu, FileOpenFailedSnafu, FileReadFailedSnafu, MaybeTls, NewCaStackSnafu,
     NewStoreBuilderSnafu, ParsePkcs12Snafu, Pkcs12Snafu, PrivateKeyParseSnafu, Result,
-    SetAlpnProtocolsSnafu, SetCertificateSnafu, SetPrivateKeySnafu, SetSniSnafu,
-    SetVerifyCertSnafu, TlsError, TlsIdentitySnafu, X509ParseSnafu,
+    SetAlpnProtocolsSnafu, SetCertificateSnafu, SetPrivateKeySnafu, SetVerifyCertSnafu, TlsError,
+    TlsIdentitySnafu, X509ParseSnafu,
 };
 
 pub const PEM_START_MARKER: &str = "-----BEGIN ";
@@ -343,12 +343,15 @@ impl TlsSettings {
         Ok(())
     }
 
-    pub fn apply_connect_configuration(&self, connection: &mut ConnectConfiguration) -> Result<()> {
+    pub fn apply_connect_configuration(
+        &self,
+        connection: &mut ConnectConfiguration,
+    ) -> std::result::Result<(), openssl::error::ErrorStack> {
         connection.set_verify_hostname(self.verify_hostname);
         if let Some(server_name) = &self.server_name {
             // Prevent native TLS lib from inferring default SNI using domain name from url.
             connection.set_use_server_name_indication(false);
-            connection.set_hostname(server_name).context(SetSniSnafu)?;
+            connection.set_hostname(server_name)?;
         }
         Ok(())
     }
diff --git a/src/http.rs b/src/http.rs
index 9c30bdea49..43bd10eb88 100644
--- a/src/http.rs
+++ b/src/http.rs
@@ -14,7 +14,6 @@ use hyper_openssl::HttpsConnector;
 use hyper_proxy::ProxyConnector;
 use rand::Rng;
 use serde_with::serde_as;
-use snafu::Error;
 use snafu::{ResultExt, Snafu};
 use std::{
     fmt,
@@ -205,19 +204,10 @@ pub fn build_tls_connector(
     let settings = tls_settings.tls().cloned();
     https.set_callback(move |c, _uri| {
         if let Some(settings) = &settings {
-            match settings.apply_connect_configuration(c) {
-                Ok(()) => (),
-                Err(error) => {
-                    error
-                        .source()
-                        .expect("was expecting to have a source in tlsError variant returned from apply_connect_configuration(). See SetSniSnafu.")
-                        .downcast_ref::<openssl::error::ErrorStack>()
-                        .expect("was expecting the source error to be of type openssl::error::ErrorStack from apply_connect_configuration(). See SetSniSnafu.");
-                }
-            }
+            settings.apply_connect_configuration(c)
+        } else {
+            Ok(())
         }
-
-        Ok(())
     });
     Ok(https)
 }
```

Reply:
This makes more sense. applied.